src/fw-logs/fw-logs-parser.cpp

```diff
@@ -1,90 +1,215 @@
 // License: Apache 2.0. See LICENSE file in root directory.
-// Copyright(c) 2019 Intel Corporation. All Rights Reserved.
-#include "fw-logs-parser.h"
+// Copyright(c) 2024 Intel Corporation. All Rights Reserved.
+
+#include <src/fw-logs/fw-logs-parser.h>
+#include <src/fw-logs/fw-string-formatter.h>
+#include <src/fw-logs/fw-logs-xml-helper.h>
+
+#include <rsutils/string/from.h>
+
 #include <regex>
-#include <sstream>
-#include "fw-string-formatter.h"
-#include "stdint.h"
+#include <fstream>
+#include <iterator>
 
-using namespace std;
 
 namespace librealsense
 {
     namespace fw_logs
     {
-        fw_logs_parser::fw_logs_parser(string xml_content)
-            : _fw_logs_formating_options(xml_content),
-            _last_timestamp(0),
-            _timestamp_factor(0.00001)
+        fw_logs_parser::fw_logs_parser( const std::string & definitions_xml )
+            : _source_to_formatting_options()
         {
-            _fw_logs_formating_options.initialize_from_xml();
+            fw_logs_xml_helper helper;
+            _source_id_to_name = helper.get_listed_sources( definitions_xml );
+            for( auto & source : _source_id_to_name )
```

Comment:
Adding const will work?
RO, right?

Reply:
Yes, can be const. Using the listed sources from the file to build the parsers map for each source, but only need the ID for each source (as a key)

Reply:
Done